components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -522,6 +529,11 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
                 LOG.debug("Retrieved the User Information:" + jwtAttributeMap);
             }
 
+            String nonce = (String) jwtAttributeMap.get(NONCE);
+            if (nonce != null && !nonce.equals(context.getProperty(NONCE))) {
```

Comment:
What if nonce is null ?

Reply:
Yes. Hope warn log would be appropriate.

Reply:
Fixed by - https://github.com/wso2-extensions/identity-outbound-auth-oidc/pull/139/commits/96c5f1dc1995eb37c0b5e6b1187eb21bc23b24dd